File: alura-cursos/ano-2023/java/bytebankHerdadoImposto/src/bytebankHerdadoImposto/ContaCorrente.java
package bytebankHerdadoImposto;

public class ContaCorrente extends Conta implements Tributavel {
	public ContaCorrente(int agencia, int numero) {
		super(agencia, numero);
	}

	@Override
	public boolean sacar(double valor) {
		double taxaTransferencia = 0.2;
		return super.sacar(valor + taxaTransferencia);
	}

	@Override
	public void depositar(double valor) {
		super.saldo += valor;
		
	}

	@Override
	public double getValorImposto() {
		double aliquota = 0.1;
		return super.saldo*aliquota;
	}

}

File: alura/ano-2023/csharp/bytebank-exception/Contas/ContaCorrente.cs
using bytebank_exception.Usuarios;

namespace bytebank_exception.Contas;
public class ContaCorrente : Conta
{
    public ContaCorrente(int agencia, int numeroConta, Cliente titular) : base(agencia, numeroConta, titular) { }

    public override bool Sacar(double valor)
    {
        if (!PodeSacar(valor))
        {
            return false;
        }

        this.Saldo -= valor + TaxaOperacao;
        return true;
    }
}
